mariadb-{LTS,devel} : updates ; boost188 ; LTS 11.8, 12.3 ; devel

BjarneDMat opened the pull request:
## Description
```
mariadb-devel          : new Port @13.0.1
mariadb-12.3           : new port @12.3.2 LTS
mariadb-11.8           : new port v11.8.6
mariadb-11.4           : upgrade to 11.4.10 ; boost188
libfmt12               : new port @12.1.0
mariadb-10.11          : upgrade to 10.11.16 ; boost188
mariadb-10.6           : upgrade to 10.6.25 ; boost188
mariadb-10.5           : upgrade to 10.5.29; boost188
```

### Instructions
In order to clone & get this to work on your own computer, this is my recommended procedure:

[https://macports.mathiesen.info/macports/](https://macports.mathiesen.info/macports/)
from gitRepos, you'll need :
1. gitFunctions.include
2. mariadb11Git

I order to get it to work, you'll need a directory to work in - eg, **MacPorts** so :

start by installing bash, curl & git:
`port -cuN install bash curl git`
and use MacPorts `bash` as your bash in Terminal

Then:
```
declare mathiesen='https://macports.mathiesen.info/macports/gitRepos'
mkdir -p MacPorts/gitRepos ; cd MacPorts/gitRepos
curl -O "${mathiesen}/gitFunctions.include"
curl -O "${mathiesen}/mariadb11Git"
chmod +x mariadb11Git
sed -i '' -E -e '/^case/s/skip/update/' gitFunctions.include
cd ..
./gitRepos/mariadb11Git sync init
```
then, you'll have to edit (as root) `/opt/local/etc/macports/sources.conf`
and insert an absolute path to your cloned `BjarneDMat:mariadb11` - in my case :
`file:///Users/macuser/MacPorts/ports/mariadb11      [nosync]`

you can test, if you've gotten everything right, by executing :
`port file mariadb-dev`

Subsequent updates can then simply be executed by :
`./gitRepos/mariadb11`
as `update` now is the default

###### Type(s)
- [x] update
- [x] enhancement
- [x] submission
- [x] bugfix

###### MariaDB issues addressed : 
[MDEV-38853](https://jira.mariadb.org/browse/MDEV-38853) Json_writer missing #if ... #endif to exclude VALIDITY_ASSERT
[MDEV-38777](https://jira.mariadb.org/browse/MDEV-38777) mariadb-test Build regression in FreeBSD - 2026Q1 releases
[MDEV-38811](https://jira.mariadb.org/browse/MDEV-38811) crash in information_schema.table_constraints when --skip-grant-tables

###### Closes
[ticket 72779](https://trac.macports.org/ticket/72779)  MariaDB: Please import MariaDB 11.8 LTS series 
[ticket 72752](https://trac.macports.org/ticket/72752)  mariadb-11.4: Wrong installation notes 
[ticket 72323](https://trac.macports.org/ticket/72323)  mariadb-10.6: fails to run due to missing dependency on lzo2 
[ticket 71702](https://trac.macports.org/ticket/71702)  mariadb-10.11 install fails on Sequoia 15.2 

###### Tested on
```
Model Identifier: MacPro5,1                 Model Identifier: Macmini6,1
macOS 15.7.1 24G231 x86_64                  macOS 10.15.8 19H2036 x86_64
Xcode 26.1.1 17B100                         Xcode 12.4 12D4e
Command Line Tools 26.1.0.0.1.1761104275    Command Line Tools 12.4.0.0.1.1610135815

Model Identifier: Mac14,3
macOS 26.2 25C56 arm64
Xcode not installed
Command Line Tools 26.2.0.0.1.1764812424

```

BjarneDM committed:
mariadb-10.5: upgrade to 10.5.29; boost188

PortGroup           deprecated 1.0
deprecated.eol_version      yes
eprecated.upstream_support  no

BjarneDM committed:
mariadb-10.6: upgrade to 10.6.25 ; boost188

BjarneDM committed:
mariadb-10.11: upgrade to 10.11.16 ; boost188

BjarneDM committed:
mariadb-11.4: upgrade to 11.4.10 ; boost188

BjarneDM committed:
mariadb-11.8: new port v11.8.6

ottok commented:
Thanks for working on this! Once merged, https://trac.macports.org/ticket/72779 can be closed.

ottok commented on databases/mariadb-11.8/Portfile:
```diff
@@ -0,0 +1,275 @@
+                        -DSYSCONFDIR:PATH=${prefix}/etc/${name_mysql} \
+                        -DMYSQL_UNIX_ADDR:PATH=${prefix}/var/run/${name_mysql}/mysqld.sock \
+                        -DMYSQL_DATADIR:PATH=${prefix}/var/db/${name_mysql} \
+                        -DDEFAULT_CHARSET:STRING=utf8 \
```
In 11.8 you probably don't want to override the upstream default charset or collation but use the latest values. To notice what is new in 11.8, you might want to browse upstream release notes or read my detailed summary at https://salsa.debian.org/mariadb-team/mariadb-server/-/blob/debian/latest/debian/mariadb-server.NEWS

ottok commented:
> from https://endoflife.date/mariadb So, basically we are in the process of going from having everything released in MacPorts to only having the LTS versions. But, what's the policy regarding deprecate & obsolete in relation to community, enterprise, extended. My suggestion is :
> 
> * deprecate when community expires
> * obsolete when enterprise expires

This https://endoflife.date/mariadb is badly written. Unfortunately https://mariadb.org/about/#maintenance-policy isn't much better either, as they write about "binary releases", but in Macports we care only about what is released as source code in upstream GitHub repository.

They should have written simply this "MariaDB 11.8 LTS was released on 04 Jun 2025 and will have security updates publicly available for 5 years until 2030."

BjarneDMat commented:
@ottok we had a discussion about this in https://github.com/macports/macports-ports/pull/27905
Presently, I'm working on-and-off (not a primary priority of mine) on deprecating and obsoleting in https://github.com/macports/macports-ports/pull/28046
Regarding endoflife, you do have the option of improving it on GitHub.

BjarneDMat commented:
```
diff --git a/databases/mariadb-11.8/Portfile b/databases/mariadb-11.8/Portfile
index fe386cf6ffa..da3a19135b6 100644
--- a/databases/mariadb-11.8/Portfile
+++ b/databases/mariadb-11.8/Portfile
@@ -102,8 +102,8 @@ if {$subport eq $name} {
                         -DSYSCONFDIR:PATH=${prefix}/etc/${name_mysql} \
                         -DMYSQL_UNIX_ADDR:PATH=${prefix}/var/run/${name_mysql}/mysqld.sock \
                         -DMYSQL_DATADIR:PATH=${prefix}/var/db/${name_mysql} \
-                        -DDEFAULT_CHARSET:STRING=utf8 \
-                        -DDEFAULT_COLLATION:STRING=utf8_general_ci \
+                        -DDEFAULT_CHARSET:STRING=utf8mb4 \
+                        -DDEFAULT_COLLATION:STRING=uca1400_ai_ci \
                         -DWITH_EMBEDDED_SERVER:BOOL=ON \
                         -DWITH_ZLIB:STRING=system \
                         -DWITH_UNIT_TESTS:BOOL=ON \
```

reneeotten commented:
if the CI fails without trouble, I will merge the PR

BjarneDMat commented:
So, 11.4 fails when downloading libfmt-12.1.0 :
```
2026-02-09T16:21:24.8982480Z CMake Error at libfmt-stamp/download-libfmt.cmake:163 (message):
2026-02-09T16:21:24.8997340Z   Each download failed!
2026-02-09T16:21:24.9005100Z 
2026-02-09T16:21:24.9005620Z     error: downloading 'https://github.com/fmtlib/fmt/releases/download/12.1.0/fmt-12.1.0.zip' failed
2026-02-09T16:21:24.9009910Z           status_code: 22
2026-02-09T16:21:24.9010250Z           status_string: "HTTP response code said error"
```
This seems to be a temporary error, because

1. there're no problems on my test computers
2. the logs for both macos-14 & macos-26 shows success
3. it works with 11.8 in the macos-15 log
```
2026-02-09T16:37:45.2776760Z    dst='/opt/local/var/macports/build/mariadb-11.8-0835226b/work/build/extra/libfmt/src/fmt-12.1.0.zip'
2026-02-09T16:37:45.2779120Z -- Using src='https://github.com/fmtlib/fmt/releases/download/12.1.0/fmt-12.1.0.zip'
2026-02-09T16:37:45.4954470Z        file='/opt/local/var/macports/build/mariadb-11.8-0835226b/work/build/extra/libfmt/src/fmt-12.1.0.zip'
2026-02-09T16:37:45.5378140Z      src='/opt/local/var/macports/build/mariadb-11.8-0835226b/work/build/extra/libfmt/src/fmt-12.1.0.zip'
```

BjarneDM committed:
libfmt12: new port @12.1.0

BjarneDM committed:
mariadb-12.3: new port @12.3.1 RC

ottok commented:
In Debian we build MariaDB 11.8 with [libfmt version 10.1.1](https://tracker.debian.org/pkg/fmtlib). You should be able to use the 10.x version for all MariaDB builds in MacPorts too. There shouldn't bee any hard requirement for version 12.

BjarneDMat commented:
@ottok 
the source files ‼️explicitly‼️ asks for `fmt-12.1.0` in `cmake/libfmt.cmake`.
& the `libfmt10` version MacPorts has isn't 10.1.0 but 10.2.1

I've added : 
```
    depends_lib-append  port:curl \
                        ... \
                        port:libfmt12
```
```
    configure.args-append \
                        ... \
                        -DLIBFMT_INCLUDE_DIR:PATH=${prefix}/include/libfmt12 \
                        ...
```
to the `mariadb-{11.4,11.8,12.3}` Portfiles

reneeotten commented on databases/mariadb-12.3/Portfile:
```diff
@@ -0,0 +1,278 @@
+# CLOCK_REALTIME
+legacysupport.newest_darwin_requires_legacy 15
+
+name                mariadb-12.3
```
why is there PREVIEW in the commit message...?

BjarneDMat replied:
That's what MariaDB calls it.
They are presently using this numbering : 
1. 12.1.X is beta-X
2. 12.2.Y is alpha-X
3. 12.3.0 is Preview ; 12.3.1 is RC-1 ...

reneeotten commented on databases/mariadb-12.3/Portfile:
```diff
@@ -0,0 +1,278 @@
+
+if {$subport eq $name} {
+
+    PortGroup           cmake 1.0
```
please use the latest version of the PG which is `1.1`

BjarneDMat replied:
done for all mariadb-{LTS}

reneeotten commented on databases/mariadb-12.3/Portfile:
```diff
@@ -0,0 +1,278 @@
+    description         Multithreaded SQL database server
+    long_description    Mariadb is a fork of the MySQL server, a multi-threaded SQL database.
+
+    cmake.out_of_source yes
```
this is the default for the `cmake 1.1` PG so please remove

reneeotten reviewed (changes requested):
@BjarneDM thanks for adding the `libfmt12` port and use that in the `mariadb-*` ports!

Ideally one should use the latest version of the `cmake` PortGroup, which is `1.1` and then the explicit `cmake.out_of_source yes` can be removed; that applies to all ports.

After that I think it should be all good to go; sorry it took a while but it looks excellent now, so almost there!

BjarneDMat commented:
w/ PortGroup cmake 1.1 in mariadb-10.5 @10.5.29 I get this:
```
$=> grep 'error:' '/opt/local/var/macports/logs/_Users_macuser_MacPorts_ports_mariadb11_databases_mariadb-10.5/mariadb-10.5/main.log'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:238:31: error: use of undeclared identifier 'named_item_expected'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:238:19: error: use of undeclared identifier 'got_name'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:271:31: error: use of undeclared identifier 'named_item_expected'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:271:19: error: use of undeclared identifier 'got_name'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:238:31: error: use of undeclared identifier 'named_item_expected'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:238:19: error: use of undeclared identifier 'got_name'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:271:31: error: use of undeclared identifier 'named_item_expected'
:info:build /opt/local/var/macports/build/mariadb-10.5-c01edc1a/work/mariadb-10.5.29/sql/my_json_writer.cc:271:19: error: use of undeclared identifier 'got_name'
```
w/ PortGroup cmake 1.0 in mariadb-10.5 @10.5.29 it compiles & installs.
As mariadb-10.5 is EOL, I don't think it's worth the time fixing the problem.

BjarneDMat commented:
I've got access to an arm64 Mac Mini tuesday 2026-02-17 CET
If someone could give me some pointers as to what II should do regarding bug-hunting before that, it' would be appreciated 🤓🙏🏻

BjarneDM committed:
mariadb-devel: new Port @12.2.2

This is the devevloper version, Thus, it is only recommended for testing and bug-hunting.
The mariadb-dev Port will be based upon the latest available <major>.\[012\].<patch> .
This will only be updated with compile breaking patches.

vuvova commented:
No, I mean, a link to a build log in your buildbot. Something like, for example, https://buildbot.mariadb.org/#/builders/708/builds/17032/steps/4/logs/stdio but the one that fails

and yes, I've seen upstream [MDEV-38853](https://jira.mariadb.org/browse/MDEV-38853), it doesn't look confirmed and approved yet, nor does the PR https://github.com/MariaDB/server/pull/4688

vuvova commented:
Thanks. Indeed, `MacPorts` basically works like `None`. No build-type-specific options are added.
The problem with that is that some options enable features (like `-fno-omit-frame-pointer`) and some options **disable** features, like `-DDBUG_OFF`. That's why you have have dbug* code enabled, which should not be happening in release builds. If you want to use `MacPorts` or `None`, you need to specify `-DDBUG_OFF` explicitly. Both in CFLAGS and CXXFLAGS.

BjarneDMat commented:
Do you want it split into one PR / Port ❓

BjarneDMat commented:
Regarding the extensive commit messages, they are a help to me during development. I do know, that they have to be short, to the point, and without the file listings when the PR is ready to be merged, so please just tell me : BjarneDMat, we are ready to merge, so clean up your commit mess 🤔🤣
